x-pack/plugin/ilm/src/main/java/org/elasticsearch/xpack/ilm/history/ILMHistoryStore.java

```diff
@@ -105,18 +138,24 @@ public void putAsync(ILMHistoryItem item) {
                 LIFECYCLE_HISTORY_INDEX_ENABLED_SETTING.getKey(), item);
             return;
         }
-        logger.trace("about to index ILM history item in index [{}]: [{}]", ILM_HISTORY_ALIAS, item);
-        ensureHistoryIndex(client, clusterService.state(), ActionListener.wrap(createdIndex -> {
-            try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
-                item.toXContent(builder, ToXContent.EMPTY_PARAMS);
-                IndexRequest request = new IndexRequest(ILM_HISTORY_ALIAS).source(builder);
-                processor.add(request);
-            } catch (IOException exception) {
-                logger.error(new ParameterizedMessage("failed to index ILM history item in index [{}]: [{}]",
-                    ILM_HISTORY_ALIAS, item), exception);
-            }
-        }, ex -> logger.error(new ParameterizedMessage("failed to ensure ILM history index exists, not indexing history item [{}]",
-            item), ex)));
+        logger.trace("queueing ILM history item for indexing [{}]: [{}]", ILM_HISTORY_ALIAS, item);
+        try (XContentBuilder builder = XContentFactory.jsonBuilder()) {
+            item.toXContent(builder, ToXContent.EMPTY_PARAMS);
+            IndexRequest request = new IndexRequest(ILM_HISTORY_ALIAS).source(builder);
+            // TODO: remove the threadpool wrapping when the .add call is non-blocking
```

Comment:
Oh, this is not intuitive/obvious. Maybe in a follow-up PR shall we rename the `add` method to `addAndMaybeExecute` or something along those lines?

Reply:
In a followup we are going to make the `add` method not block on execution (but still block on adding to the queue). I opened #50440 for this (and I'll link it in the comment in the code)

Reply:
Brill, thanks @dakrone